py-polars/polars/mp_check.py

```diff
@@ -0,0 +1,16 @@
+import multiprocessing as mp
+
+# Set python multiprocessing start method to "spawn", if undefined
+# or raise ImportError, if defined, but not set to "spawn".
+# Python multiprocessing with start method "fork" can not be used
+# as forking does not copy threads (Polars spawns multiple threads
+# for its rayon threadpool), resulting in locks that can't be
+# be released, resulting in a hanging python interpreter.
+mp_method = mp.get_start_method(allow_none=True)
+if not mp_method:
+    mp.set_start_method("spawn")
+elif mp_method != "spawn":
+    raise ImportError(
+        "Polars only works with python multithreading method set to spawn: "
```

Comment:
Should multithreading be multiprocessing?

Reply:
Yeah, will force push. It also seems to work with "forkserver" (besides "spawn").